### ISSUE TITLE
length score?

### ISSUE THREAD

**Issue by Jon-Zbw** [NVlabs/GDPO#2]
```
elif adv_estimator == 'gdpo':
        ## only handle two reward now
        token_level_scores_correctness = data.batch['token_level_scores_correctness']
        token_level_scores_format = data.batch['token_level_scores_format']
        
        # shared variables 
        index = data.non_tensor_batch['uid']
        responses = data.batch['responses']
        response_length = responses.size(-1)
        attention_mask = data.batch['attention_mask']
        response_mask = attention_mask[:, -response_length:]

        ## handle correctness first
        correctness_normalized_score, _ = core_algos.compute_grpo_outcome_advantage(token_level_rewards=token_level_scores_correctness,
                                                                        eos_mask=response_mask,
                                                                        index=index)

        ## handle format now
        format_normalized_score, _ = core_algos.compute_grpo_outcome_advantage(token_level_rewards=token_level_scores_format,
                                                                        eos_mask=response_mask,
                                                                        index=index) 
        
        new_advantage = correctness_normalized_score + format_normalized_score

        advantages = masked_whiten(new_advantage, response_mask) * response_mask

        data.batch['advantages'] = advantages
        data.batch['returns'] = advantages
```
So length_score was no use in the end?

**Comment by FangxuLiu** [comment]
The paper mentions that normalization will be performed again within the batch, but I don't seem to see this process described. Is this correct?

**Comment by nbasyl** [comment]
@Jon-Zbw Yes, for the ToolRL experiments, we follow the original setup and do not use the length_score.
@FangxuLiu masked_whiten is the batch normalization function provided by verl; for the TRL implementation, I need to implement the batch normalization manually.
